### ISSUE TITLE
Highlighting is randomly broken

### ISSUE THREAD

**Issue by evilhamsterman** [VSChina/vscode-ansible#181]
## Environment
- Extension version: 0.4.0
- OS Type: MacOS

## Summary
After fixing the issues with nested lists of dictionaries in #168 now the issue is happening with just general attributes of dictionaries, when it fails seems random. One specific case I found is if a hostname in an inventory has a - in the name.

## Reproduce steps
Create a task or an inventory file. Example the last line `backup` is not highlighted but it is if you move it up a line

```YAML
- name: Point repos to local mirror  on {{ inventory_hostname }}
  become: yes
  template:
    src: templates/sources.list.j2
    dest: /etc/apt/sources.list
    owner: root
    group: root
    mode: "u=rw,g=r,o=r"
    backup: yes
```

## Expected Results
Everything is highlighted if backup isn't the last line
![screen shot 2018-09-14 at 1 33 10 pm](https://user-images.githubusercontent.com/59560/45573899-e5498680-b822-11e8-8c1b-e82a624c9dde.png)

Everything is highlighted
![screen shot 2018-09-14 at 1 37 09 pm](https://user-images.githubusercontent.com/59560/45574015-5426df80-b823-11e8-8c0a-7537f14daf1d.png)

## Actual Results
![screen shot 2018-09-14 at 1 32 56 pm](https://user-images.githubusercontent.com/59560/45573915-f2ff0c00-b822-11e8-9b16-5f9618c86749.png)
![screen shot 2018-09-14 at 1 37 02 pm](https://user-images.githubusercontent.com/59560/45574018-58eb9380-b823-11e8-8e4f-61e594314bae.png)

**Comment by yungezz** [comment]
the first case could be fixed by PR #182 , it's due to matching attributes aggressive end. While the second case is actually caused by the `-` in variable name, quick workaround is remote it, `host-name` to `hostname`. Fixing of it will need touch  base syntax regex expression of ansible, it need be careful, and verbose testing.

**Comment by yungezz** [comment]
we  don't have plan for fixing second issue for now.  the first issue was fixed and release. close the issue for now.
